rhoas-client/src/main/java/com/redhat/service/bridge/rhoas/RhoasClientImpl.java

```diff
@@ -50,11 +50,34 @@ public RhoasClientImpl(KafkasMgmtV1Client mgmtClient, KafkaInstanceAdminClient i
                 .onFailure().transform(f -> logAndWrapFailure("Error when deleting service account with id " + id, f));
     }
 
+    @Override
+    public Uni<Topic> getTopic(String topicName) {
+        return instanceClient.getTopic(topicName)
+                .onItem().invoke(t -> LOG.info("Retrieved topic '{}'", t.getName()))
+                .onFailure().transform(f -> logAndWrapFailure("Error when retrieving topic " + topicName, f));
+    }
+
     @Override
     public Uni<Topic> createTopic(NewTopicInput newTopicInput) {
         return instanceClient.createTopic(newTopicInput)
                 .onItem().invoke(t -> LOG.info("Created topic '{}'", t.getName()))
-                .onFailure().transform(f -> logAndWrapFailure("Error when creating topic " + newTopicInput.getName(), f));
+                .onFailure()
+                .recoverWithUni(f -> handleCreateTopicFailure(newTopicInput.getName(), f));
+    }
+
+    private Uni<Topic> handleCreateTopicFailure(String topicName, Throwable f) {
+        if (!(f instanceof AppServicesException)) {
+            RhoasClientException rhoasClientException = logAndWrapFailure("Error when creating topic " + topicName, f);
+            return Uni.createFrom().failure(rhoasClientException);
+        }
+        AppServicesException exception = (AppServicesException) f;
+        if (exception.getStatusCode() == HttpStatus.SC_CONFLICT) {
+            LOG.info("Topic {} has already been created", topicName);
+            return getTopic(topicName);
```

Comment:
I call our "public" `getTopic(topicName)` method to support the potential for it too to fail.